pepper-apis/src/main/java/org/broadinstitute/ddp/util/PubSubEmulator.java

```diff
@@ -149,6 +153,27 @@ public static void shutdown() throws IOException {
     }
 
     public static synchronized boolean hasStarted() {
-        return pubsubProcess != null;
+        // We might be running a Docker based emulator
+        if (isExternalEmulatorRunning()) {
+            LOG.info("An emulator already running at " + EMULATOR_HOST);
+            return true;
+        } else {
+            return pubsubProcess != null;
+        }
+    }
+
+    private static boolean isExternalEmulatorRunning() {
+        try (
+                CloseableHttpClient httpClient = HttpClients.createDefault();
+                CloseableHttpResponse response =
+                        httpClient.execute(new HttpGet("http://" + EMULATOR_HOST));
+        ) {
+            return response.getStatusLine().getStatusCode() == 200;
+        } catch (IOException e) {
+            String msg = "There was problem initializing CloseableHttpClient";
+            LOG.error(msg, e);
+            throw new DDPException(msg, e);
+        }
+
```

Comment:
If emulator is already running, we don't start a new one. We setup our own in CircleCi test environment.